### ISSUE TITLE
Triangle License

### ISSUE THREAD

**Issue by stuaxo** [glumpy/glumpy#118]
Triangles license says
 "Please note that although Triangle is freely available, it is copyrighted by the author and may not be sold or included in commercial products without a license. "

now... anything I make is unlikely to be commercial, but who knows.   It's more straightforward to stick with standard licenses like BSD, LGPL etc.  - non commercial licenses seem to be an issue if I were to ever get paid.

Is it possible to use Glumpy without Triangle ?   How reliant on Triangle is it, and what would be missing without it ?

**Comment by rougier** [comment]
Good point. There is actually two licenses involved. The one from the C library and the one from the binding. The original license reads:

```
These programs may be freely redistributed under the condition that the
copyright notices (including the copy of this notice in the code comments
and the copyright notice printed when the `-h' switch is selected) are
not removed, and no compensation is received.  Private, research, and
institutional use is free.  You may distribute modified versions of this
code UNDER THE CONDITION THAT THIS CODE AND ANY MODIFICATIONS MADE TO IT
IN THE SAME FILE REMAIN UNDER COPYRIGHT OF THE ORIGINAL AUTHOR, BOTH
SOURCE AND OBJECT CODE ARE MADE FREELY AVAILABLE WITHOUT CHARGE, AND
CLEAR NOTICE IS GIVEN OF THE MODIFICATIONS.  Distribution of this code as
part of a commercial system is permissible ONLY BY DIRECT ARRANGEMENT
WITH THE AUTHOR.  (If you are not directly supplying this code to a
customer, and you are instead telling them how they can obtain it for
free, then you are not required to make any arrangement with me.)
```

And there might a license for the binding but I dit not find it.



To answer your question about using grumpy without triangle, it is possible but you will lack everything that is related to polygon triangluation (such as the tiger demo). The alternative would be to find another library that do more or less the same. There are a lot around but not all of them offer Python binding.

**Comment by stuaxo** [comment]
So from reading that, it seems like you can distribute triangle and it's code.

I *do* believe devs should be get paid (and indeed I do).  OTOH, Graphics stuff isn't my job - if I go and do some VJing and get paid using Glumpy + Triangle, should I be seeking out a license from the Triangle authors...

**Comment by QuLogic** [comment]
On PyPI, the triangle wrapper metadata states it is LGPL. However, the (L)GPL doesn't allow further restrictions (such as no-fees-allowed), so I don't think the wrapper is consistent itself, and I would not rely on it.

Matplotlib, for example, uses triangulation from qhull that maybe could be used instead. SciPy also has a wrapper for qhull.

**Comment by rougier** [comment]
I considered qhull at some point but it does not offer all the necessary functions (maybe it has changed since then).

**Comment by stuaxo** [comment]
Are they features that could be reasonably requested from qhull, or is it something fundamental which makes it look unimplementable there ?

**Comment by rougier** [comment]
I'm not sure I remember exactly what was the limitation. Maybe the possibility to specify holes in the geometry.
